src/common/metrics/ngx_wa_histogram.c

```diff
@@ -205,8 +242,20 @@ ngx_wa_metrics_histogram_record(ngx_wa_metrics_t *metrics, ngx_wa_metric_t *m,
     h = m->slots[slot].histogram;
     h->sum += n;
 
-    b = histogram_bin(metrics, h, n, &m->slots[slot].histogram);
-    b->count += 1;
+    switch (h->h_type) {
+    case NGX_WA_HISTOGRAM_LOG2:
+        b = histogram_log2_bin(metrics, h, n, &m->slots[slot].histogram);
+        break;
+    case NGX_WA_HISTOGRAM_CUSTOM:
+        b = histogram_custom_bin(h, n);
+        break;
+    default:
+        return NGX_ERROR;
+    }
+
+    if (b) {
```

Comment:
So if `b` is `NULL` (which can only happen with custom bins?), what does it mean? A comment maybe?

Reply:
`b` can't really be `NULL`. I added the check anticipating that the compiler would complain about dereferencing a pointer that *theoretically* can be `NULL` (e.g., passing a histogram with `n_bins=0` to `histogram_custom_bin` would make it return a `NULL` pointer).

I'll test removing the check.

Reply:
Then an assertion is more appropriate; we also might want to add a similar assertion to the custom_bin function, which initializes `b = NULL` and then runs the loop. If we add an assertion before the final `return b` we show the developer that the function cannot return `NULL`.

Reply:
Good point. I'll use assertions to communicate the validity of the pointers.